server/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java

```diff
@@ -108,7 +108,7 @@ public void insert(InsertRowPlan insertRowPlan) {
       Object value = insertRowPlan.getValues()[i];
       memSize += MemUtils.getRecordSize(insertRowPlan.getSchemas()[i].getType(), value);
 
-      write(insertRowPlan.getDeviceId(), insertRowPlan.getMeasurements()[i],
+      write(insertRowPlan.getDeviceId().getFullPath(), insertRowPlan.getMeasurements()[i],
```

Comment:
I think we can use PartialPath directly here. There is no need to transfer it to String. If so, you need to change the type of `memTableMap` in `AbstractMemTable` to `Map<PartialPath, Map<String, IWritableMemChunk>> memTableMap`.
And by the way, i think you should rewrite the both hashCode and compareTo methods in PartialPath. We can compare each string in string array`nodes` instead of calling the fullPath() method which will retain it in memory until the PartialPath is cleaned by GC.